client/src/Pages/Dallas.js

```diff
@@ -0,0 +1,226 @@
+  };
+
+  const handleAltChange = (event) => {
+    const alt = event.target.value;
+    setPost((post) => ({ ...post, alt }));
+  };
+
+  const handleDishChange = (event) => {
+    const dish = event.target.value;
+    setPost((post) => ({ ...post, dish }));
+  };
+
+  const handleRestaurantChange = (event) => {
+    const restaurant = event.target.value;
+    setPost((post) => ({ ...post, restaurant }));
+  };
+
+  const handleContentChange = (event) => {
+    const content = event.target.value;
+    setPost((post) => ({ ...post, content }));
+  };
+
+  const handleCityChange = (event) => {
+    const city = event.target.value;
+    setPost((post) => ({ ...post, city }));
+  };
+
+  const handleDateChange = (event) => {
+    const date = event.target.value;
+    setPost((post) => ({ ...post, date }));
+  };
+
+  function getRestaurants(e) {
+    e.preventDefault();
+    fetch("/api/location-search", {
+      method: "GET",
+      headers: {"Content-Type": "application/json",},
+    }) .then((response) => response.json())
+    .then((data) => {
+      console.log("data check line 13 ", data.businesses[0]);
+      let fetchRestaurant = {
+        alias: data.businesses[0].alias,
+        name: data.businesses[0].name,
+        address: data.businesses[0].location.display_address
+      }
+      setRestaurants(fetchRestaurant);
+      console.log("checking restaurants line 24 ", restaurants)
+    }) .catch((err) => console.error(err))
+  }
+
+
+  //A function to handle the post request
+  const makePost = (newPost) => {
+    return fetch("/api/blogposts", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newPost),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("From the post ", data);
+        props.savePost(data);
+      });
+  };
+
+  //a function to handle the Update request
+  const updatePost = (existingPost) => {
+    return fetch(`/api/blogposts/${existingPost.id}`, {
+      method: "PUT",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(existingPost),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("From put request ", data);
+        props.savePost(data);
+      });
+  };
+
+  // Than handle submit function now needs the logic for the update scenario
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (post.id) {
+      updatePost(post);
+    } else {
+      makePost(post);
+    }
+  };
+
+  return (
+    <div>
+      <h1>Dallas Ventures Heeere</h1>
+
+      <form onSubmit={handleSubmit}>
+        <fieldset>
+          <label>Username</label>
+          <input
+            type="text"
+            id="add-user-name"
+            placeholder="Username"
+            required
+            value={post.username}
+            onChange={handleUsernameChange}
```

Comment:
There's a good example on https://reactjs.org/docs/forms.html for having a generic handler for multiple inputs. That way you can just do `onChange={handleAnyChange}`

Reply:
k I'll check it out